### PR TITLE
[SET] Fix transition restart

### PR DESCRIPTION
**Pull request by piaskowyk**
## Summary

This PR fixes the issue with shared element transition restart. Previously, there was a chance of the app crashing or a needed component disappearing. The purpose of this PR is to prevent app crashes and ensure a proper end state for the transition.

## Test plan

Run example `TransitionRestart.tsx` example - available in this PR

### CODE REVIEW

File: src/reanimated2/layoutReanimation/animationsManager.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ function createLayoutAnimationManager() {
       if (!value) {
         return;
       }
-      stopObservingProgress(tag, value, true);
+      stopObservingProgress(tag, value);
```
Comment (tjzel) [inline src/reanimated2/layoutReanimation/animationsManager.ts:98]: Previously there was `true` here passed, now we have default `false` - is that intended?

**Review by tjzel — COMMENTED** [review]
I don't understand all the changes but it seems ok. I'd like to improve code quality for some of the places that I mentioned.

**Review by kacperkapusciak — APPROVED** [review]
I didn't understand anything from this PR so I've QA it instead

On `main` the app crashes when a Shared Transition is interrupted by the user or programmatically

With this PR applied the transition snaps to the final stage without crashing 

lgtm :shipit:

File: src/reanimated2/layoutReanimation/animationsManager.ts (hunk 1 above)
Comment (piaskowyk) [reply src/reanimated2/layoutReanimation/animationsManager.ts:99]: Yes, this is intended. This parameter is related to Layout Aniamations. We don't want to clear Layout Animation configuration here.

File: src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts [hunk 2]
```diff
@@ -158,14 +171,29 @@ function createProgressTransitionRegister() {
       }
     },
     onTransitionEnd: (removeViews = false) => {
+      if (currentTransitions.size === 0) {
+        toRemove.clear();
```
[the remark this answers is not in the document]
Comment (piaskowyk) [reply src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts:176]: yes, this is intentionally

**Review by tjzel — APPROVED** [review]
🪨

**Pull request merged**